### ISSUE TITLE
Typing errors with updated @types/react-router@5.0.0

### ISSUE THREAD

**Issue by danielsogl** [CharlesStover/use-react-router#15]
The `@types/react-router` was updated to match the latest `react-router` release 5.0.0.
Since that the following line of code is broken.

```ts
const routerContext: RouteComponentProps<T> = useContext(__RouterContext);
```

Error message:
```
const routerContext: RouteComponentProps<T, StaticContext, any>
Type 'RouteComponentProps<{}, StaticContext, any>' is not assignable to type 'RouteComponentProps<T, StaticContext, any>'.
  Type '{}' is not assignable to type 'T'.ts
```

**Comment by quisido** [comment]
I'm unable to reproduce this.

`package.json`
```
{
  "name": "use-react-router_",
  "version": "0.1.0",
  "private": true,
  "dependencies": {
    "@types/jest": "24.0.13",
    "@types/node": "12.0.2",
    "@types/react": "16.8.18",
    "@types/react-dom": "16.8.4",
    "@types/react-router": "^5.0.1",
    "@types/react-router-dom": "^4.3.3",
    "react": "^16.8.6",
    "react-dom": "^16.8.6",
    "react-router": "^5.0.0",
    "react-router-dom": "^5.0.0",
    "react-scripts": "3.0.1",
    "typescript": "3.4.5",
    "use-react-router": "^1.0.6"
  },
  "scripts": {
    "start": "react-scripts start",
    "build": "react-scripts build",
    "test": "react-scripts test",
    "eject": "react-scripts eject"
  },
  "eslintConfig": {
    "extends": "react-app"
  },
  "browserslist": {
    "production": [
      ">0.2%",
      "not dead",
      "not op_mini all"
    ],
    "development": [
      "last 1 chrome version",
      "last 1 firefox version",
      "last 1 safari version"
    ]
  }
}
```

`index.tsx`
```
import React from 'react';
import ReactDOM from 'react-dom';
import { BrowserRouter } from 'react-router-dom';
import useReactRouter from 'use-react-router';

const App: React.FC = () => {
  const route = useReactRouter();
  return (
    <div className="App">
      <header className="App-header">
        <pre style={{ textAlign: 'left' }}>{JSON.stringify(route)}</pre>
      </header>
    </div>
  );
}

ReactDOM.render(
  <BrowserRouter>
    <App />
  </BrowserRouter>,
  document.getElementById('root')
);
```

I can `yarn build` and `yarn start` without the error you provided. Can you provide a minimal reproduction? Was it perhaps patched in `@types/react-router@5.0.1`?

**Comment by quisido** [comment]
While I couldn't reproduce this, the patch in 1.0.7 hard-sets the router context to have `RouteComponentProps<any>`, which should resolve the error in the OP.
